runtime/interface.cpp

```diff
@@ -2373,7 +2373,15 @@ void init_runtime_environment(php_query_data *data, void *mem, size_t mem_size)
   init_superglobals(data);
 }
 
+static void free_shutdown_functions() {
+  for (std::size_t i = 0; i < shutdown_functions_count; ++i) {
+    shutdown_functions[i].~shutdown_function_type();
+  }
+  shutdown_functions_count = 0;
+}
+
 void free_runtime_environment() {
+  free_shutdown_functions();
```

Comment:
I think it's better to move this call into `free_interface_lib`
because this all belongs to interface.cpp and in `init_interface_lib` we reset some shutdown functions related variables, you can check here:
https://github.com/VKCOM/kphp/blob/master/runtime/interface.cpp#L2263

Reply:
done